Sources/MusicXML/Complex Types/MusicData.swift

```diff
@@ -89,55 +89,32 @@ extension MusicData: Codable {
             return try container.decode(T.self, forKey: key)
         }
 
-        // FIXME: Attempt to escape do-try-catch hell
-        do {
+        if container.contains(.note) {
```

Comment:
Oh wow. How have I not seen / thought of this? This is both cleaner and (I presume) more performant. :+1:

Reply:
I wonder if there's a way to mass find-replace other instances like this.

Reply:
I tried to make all the associated music data types conform a protocol type and dynamically initialize the enum based one the associated type. 

The biggest problem is that
`func decode<T>(_ type: T.Type, from data: Data) throws -> T where T : Decodable`
does not accept a protocol type, so I cannot tell it which type to decode dynamically.